clippy_lints/src/literal_representation.rs

```diff
@@ -82,6 +82,25 @@ declare_clippy_lint! {
     "integer literals with digits grouped inconsistently"
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Warns if hexadecimal or binary literals are not grouped
+    /// by nibble or byte.
+    ///
+    /// **Why is this bad?** Negatively impacts readability.
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Example:**
+    ///
+    /// ```rust
+    /// let x: u32 = 0xFFF_FFF;
+    /// let y: u8 = 0b01_011_101;
+    /// ```
+    pub UNUSUAL_BYTE_GROUPING,
```

Comment:
```suggestion
    pub UNUSUAL_BYTE_GROUPINGS,
```

Make sure to run `cargo dev update_lints` and to update the reference files after this change.